### PR TITLE
Enable pre-commit for python

### PR DESCRIPTION
**Pull request by huazhouwang**
## What does this implement/fix? Explain your changes.
* enable pre-commit for python
  * only some basic features are enabled now
  * flake8 and bandit will be enabled later

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
…

## Pull request type
 
_Put an `x` in the boxes that apply_
- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
…

## Any other comments?
After this PR merged, everyone should do please:
* pip install pre-commit
* pre-commit install

### CODE REVIEW

**Comment by huazhouwang** [comment]
@shuaiLiWang 我加入了 pre-commit ，有空帮忙看下 PR

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    -   id: end-of-file-fixer
+        exclude: /(migrations|_pb2)/
+        files: .*.py$
+    -   id: trailing-whitespace
+        exclude: /(migrations|_pb2)/
+        files: .*.py$
+    -   id: file-contents-sorter
+        files: ^(requirements.*.txt)$
+
+#-   repo: https://github.com/PyCQA/bandit
```
Comment (danielsocials) [inline .pre-commit-config.yaml:42]: 删除无用代码

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        exclude: /(migrations|_pb2)/
+
+-   repo: https://github.com/psf/black
+    rev: 19.10b0
+    hooks:
+    -   id: black
+        language_version: python3
+        exclude: /(\\.eggs|\\.git|\\.hg|\\.mypy_cache|\\.nox|\\.tox|\\.venv|_build|buck-out|build|dist|migrations|_pb2)/
+
+#-   repo: https://gitlab.com/pycqa/flake8
```
Comment (danielsocials) [inline .pre-commit-config.yaml:23]: 删除无用代码

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+repos:
+
+-   repo: https://github.com/asottile/seed-isort-config
```
Comment (danielsocials) [inline .pre-commit-config.yaml:3]: seed-isort-config is DEPRECATED?

File: android/app/generate_strings.py [hunk 4]
```diff
@@ -7,35 +7,108 @@
 # the package `requests`, and the external commands `xgettext` and `msgfmt`.
 
 import argparse
```
Comment (danielsocials) [inline android/app/generate_strings.py:9]: wrong change file?

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    hooks:
+    -   id: seed-isort-config
+
+-   repo: https://github.com/pre-commit/mirrors-isort
+    rev: v5.1.4
+    hooks:
+    -   id: isort
+        args: ["--profile", "black", "--filter-files"]
+        language_version: python3
+        exclude: /(migrations|_pb2)/
```
Comment (danielsocials) [inline .pre-commit-config.yaml:14]: migration|_pb2?

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+-   repo: https://github.com/psf/black
```
Comment (danielsocials) [inline .pre-commit-config.yaml:16]: black是否会强制format代码，style是否可以适配存在的electrum的风格？考虑到已经存在大量electrum/trezor代码，代码风格这里先只做检查，让coder自己选择去format

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+#-   repo: https://gitlab.com/pycqa/flake8
+#    rev: 3.8.3
+#    hooks:
+#    -   id: flake8
+#        language_version: python3
+#        exclude: /(migrations|_pb2)/
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.1.0
```
Comment (danielsocials) [inline .pre-commit-config.yaml:31]: 为什么不选择最新版本？

File: android/app/build.gradle [hunk 8]
```diff
@@ -15,7 +15,7 @@ android {
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         def ecVersion = null
         file("$REPO_ROOT/electrum/version.py").eachLine {
-            def matcher = it =~ / *ELECTRUM_VERSION *= *'([0-9.]+)'.*/
+            def matcher = it =~ / *ELECTRUM_VERSION *= *"([0-9.]+)".*/
```
Comment (danielsocials) [inline android/app/build.gradle:18]: 这里具体是fix了什么问题？

**Review by danielsocials — CHANGES_REQUESTED** [review]

**Comment by huazhouwang** [comment]
ack, 有些考虑不周的地方，我修改下

**Pull request closed without merging**